### PR TITLE
TECH-3901: Rename Kafka -> EbKafka

### PR DESCRIPTION
**Pull request by vdinovi**
Based on the 0.5.4 release

Renames Kafka top level namespace to EbKafka. This allows us to run this old version of ruby-kafka for Elephant bus and the newer version of ruby-kafka for Racecar within the same repo. This is a short term solution until we are able to convert all elephant bus consumers to racecar.

### CODE REVIEW

**Comment by vdinovi** [comment]
```
bin/setup
...
bundle exec rake spec
...
Finished in 1.02 seconds (files took 0.39215 seconds to load)
119 examples, 0 failures
```

File: lib/kafka/protocol/api_versions_response.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def to_s
         end
 
         def inspect
-          "#<Kafka api version #{to_s}>"
+          "# <EbKafka api version #{to_s}>"
```
Comment (echistyakov) [inline lib/kafka/protocol/api_versions_response.rb]: Extra whitespace

File: lib/kafka/protocol/message.rb [hunk 2]
```diff
@@ -49,7 +49,7 @@ def compressed?
         @codec_id != 0
       end
 
-      # @return [Array<Kafka::Protocol::Message>]
+      # @return [Array <EbKafka::Protocol::Message>]
```
Comment (echistyakov) [inline lib/kafka/protocol/message.rb]: Whitespace

File: .gitignore [hunk 3]
```diff
@@ -10,3 +10,4 @@
 .env
 *.log
 *.swp
+.idea
```
Comment (jebentier) [inline .gitignore]: This should be in your global git ignore, not added to repo specific git ignores

File: spec/functional/client_spec.rb [hunk 4]
```diff
@@ -54,8 +54,8 @@
       Timecop.freeze(now) do
         kafka.deliver_message("yolo", topic: topic, key: "xoxo", partition: 0)
       end
-    }.to raise_exception(Kafka::DeliveryFailed) {|exception|
-      expect(exception.failed_messages).to eq [Kafka::PendingMessage.new("yolo", "xoxo", topic, 0, nil, now)]
+    }.to raise_exception(EbKafka::DeliveryFailed) {|exception|
+      expect(exception.failed_messages).to eq  [EbKafka::PendingMessage.new("yolo", "xoxo", topic, 0, nil, now)]
```
Comment (echistyakov) [inline spec/functional/client_spec.rb]: Whitespace

File: spec/producer_spec.rb [hunk 5]
```diff
@@ -136,8 +136,8 @@
 
       producer.produce("hello1", topic: "greetings", partition: 0)
 
-      expect { producer.deliver_messages }.to raise_error(Kafka::DeliveryFailed) {|exception|
-        expect(exception.failed_messages).to eq [Kafka::PendingMessage.new("hello1", nil, "greetings", 0, nil, now)]
+      expect { producer.deliver_messages }.to raise_error(EbKafka::DeliveryFailed) {|exception|
+        expect(exception.failed_messages).to eq  [EbKafka::PendingMessage.new("hello1", nil, "greetings", 0, nil, now)]
```
Comment (echistyakov) [inline spec/producer_spec.rb]: Whitespace

File: spec/producer_spec.rb [hunk 6]
```diff
@@ -171,12 +171,12 @@
...
     it "handles when there's a connection error when fetching topic metadata" do
-      allow(cluster).to receive(:get_leader).and_raise(Kafka::ConnectionError)
+      allow(cluster).to receive(:get_leader).and_raise(EbKafka::ConnectionError)
 
       producer.produce("hello1", topic: "greetings", partition: 0)
 
-      expect { producer.deliver_messages }.to raise_error(Kafka::DeliveryFailed) {|exception|
-        expect(exception.failed_messages).to eq [Kafka::PendingMessage.new("hello1", nil, "greetings", 0, nil, now)]
+      expect { producer.deliver_messages }.to raise_error(EbKafka::DeliveryFailed) {|exception|
+        expect(exception.failed_messages).to eq  [EbKafka::PendingMessage.new("hello1", nil, "greetings", 0, nil, now)]
```
Comment (echistyakov) [inline spec/producer_spec.rb]: Whitespace

**Review by jebentier — APPROVED** [review]
Nice and simple global rename.  Looks good to me, just one comment about your addition to the `.gitignore`

File: spec/producer_spec.rb [hunk 7]
```diff
@@ -191,12 +191,12 @@
...
     it "handles when there's a connection error when refreshing cluster metadata" do
-      allow(cluster).to receive(:refresh_metadata_if_necessary!).and_raise(Kafka::ConnectionError)
+      allow(cluster).to receive(:refresh_metadata_if_necessary!).and_raise(EbKafka::ConnectionError)
 
       producer.produce("hello1", topic: "greetings", partition: 0)
 
-      expect { producer.deliver_messages }.to raise_error(Kafka::DeliveryFailed) {|exception|
-        expect(exception.failed_messages).to eq [Kafka::PendingMessage.new("hello1", nil, "greetings", 0, nil, now)]
+      expect { producer.deliver_messages }.to raise_error(EbKafka::DeliveryFailed) {|exception|
+        expect(exception.failed_messages).to eq  [EbKafka::PendingMessage.new("hello1", nil, "greetings", 0, nil, now)]
```
Comment (echistyakov) [inline spec/producer_spec.rb]: W

File: spec/producer_spec.rb [hunk 8]
```diff
@@ -234,13 +234,13 @@
...
         producer.produce("hello1", topic: "greetings", partition: 0)
-        expect { producer.deliver_messages }.to raise_error(Kafka::DeliveryFailed)
+        expect { producer.deliver_messages }.to raise_error(EbKafka::DeliveryFailed)
       end
 
       event = events.last
 
       expect(event.payload[:topic]).to eq "greetings"
-      expect(event.payload[:exception]).to eq ["Kafka::UnknownTopicOrPartition", "hello"]
+      expect(event.payload[:exception]).to eq [ "EbKafka::UnknownTopicOrPartition", "hello"]
```
Comment (echistyakov) [inline spec/producer_spec.rb]: W

File: spec/producer_spec.rb [hunk 9]
```diff
@@ -258,13 +258,13 @@
...
         producer.produce("hello1", topic: "greetings", partition: 0)
-        expect { producer.deliver_messages }.to raise_error(Kafka::DeliveryFailed)
+        expect { producer.deliver_messages }.to raise_error(EbKafka::DeliveryFailed)
       end
 
       event = events.last
 
       expect(event.payload[:topic]).to eq "greetings"
-      expect(event.payload[:exception]).to eq ["Kafka::UnknownTopicOrPartition", "Kafka::UnknownTopicOrPartition"]
+      expect(event.payload[:exception]).to eq [ "EbKafka::UnknownTopicOrPartition",  "EbKafka::UnknownTopicOrPartition"]
```
Comment (echistyakov) [inline spec/producer_spec.rb]: W

File: lib/kafka/broker.rb [hunk 10]
```diff
@@ -64,8 +64,8 @@ def list_offsets(**options)
 
     # Produces a set of messages to the broker.
     #
-    # @param (see Kafka::Protocol::ProduceRequest#initialize)
-    # @return [Kafka::Protocol::ProduceResponse]
+    # @param (see EbKafka::Protocol::ProduceRequest#initialize)
+    # @return  [EbKafka::Protocol::ProduceResponse]
```
Comment (echistyakov) [inline lib/kafka/broker.rb]: Whitespaces above

File: lib/kafka/client.rb [hunk 11]
```diff
@@ -205,7 +205,7 @@ def deliver_message(value, key: nil, topic:, partition: nil, partition_key: nil,
     #   be in a message set before it should be compressed. Note that message sets
     #   are per-partition rather than per-topic or per-producer.
     #
-    # @return [Kafka::Producer] the Kafka producer.
+    # @return  [EbKafka::Producer] the EbKafka producer.
```
Comment (echistyakov) [inline lib/kafka/client.rb]: W

File: .gitignore (hunk 3 above)
Comment (vdinovi) [reply .gitignore]: Oh yea, I forgot that existed. Will do.

**Review by echistyakov — COMMENTED** [review]
A lot of extra whitespaces inserted with the refactor, I marked some of them

**Comment by vdinovi** [comment]
Good catch on the whitespace -- artifact of find-and-replace. Will comb through the changes and remove them.

File: lib/kafka/client.rb [hunk 12]
```diff
@@ -371,7 +371,7 @@ def consumer(group_id:, session_timeout: 30, offset_commit_interval: 10, offset_
     #   response message set. Default is 1 MB. You need to set this higher if you
     #   expect messages to be larger than this.
     #
-    # @return [Array<Kafka::FetchedMessage>] the messages returned from the broker.
+    # @return [Array <EbKafka::FetchedMessage>] the messages returned from the broker.
```
Comment (echistyakov) [inline lib/kafka/client.rb]: One more here

File: lib/kafka/client.rb [hunk 13]
```diff
@@ -462,9 +462,9 @@ def each_message(topic:, start_from_beginning: true, max_wait_time: 5, min_bytes
     # @param timeout [Integer] a duration of time to wait for the topic to be
     #   completely created.
     # @param config [Hash] topic configuration entries. See
-    #   [the Kafka documentation](https://kafka.apache.org/documentation/#topicconfigs)
+    #   [the EbKafka documentation](https://kafka.apache.org/documentation/#topicconfigs)
     #   for more information.
-    # @raise [Kafka::TopicAlreadyExists] if the topic already exists.
+    # @raise  [EbKafka::TopicAlreadyExists] if the topic already exists.
```
Comment (echistyakov) [inline lib/kafka/client.rb]: And here

File: lib/kafka/client.rb [hunk 14]
```diff
@@ -616,9 +616,9 @@ def build_ssl_context(ca_cert_file_path, ca_cert, client_cert, client_cert_key,
           key: OpenSSL::PKey.read(client_cert_key)
         )
       elsif client_cert && !client_cert_key
-        raise ArgumentError, "Kafka client initialized with `ssl_client_cert` but no `ssl_client_cert_key`. Please provide both."
+        raise ArgumentError,  "EbKafka client initialized with `ssl_client_cert` but no `ssl_client_cert_key`. Please provide both."
       elsif !client_cert && client_cert_key
-        raise ArgumentError, "Kafka client initialized with `ssl_client_cert_key`, but no `ssl_client_cert`. Please provide both."
+        raise ArgumentError,  "EbKafka client initialized with `ssl_client_cert_key`, but no `ssl_client_cert`. Please provide both."
```
Comment (echistyakov) [inline lib/kafka/client.rb]: W

File: lib/kafka/client.rb [hunk 15]
```diff
@@ -488,13 +488,13 @@ def delete_topic(name, timeout: 30)
 
     # Describe the configuration of a topic.
     #
-    # Retrieves the topic configuration from the Kafka brokers. Configuration names
-    # refer to [Kafka's topic-level configs](https://kafka.apache.org/documentation/#topicconfigs).
+    # Retrieves the topic configuration from the EbKafka brokers. Configuration names
+    # refer to  [EbKafka's topic-level configs](https://kafka.apache.org/documentation/#topicconfigs).
```
Comment (echistyakov) [inline lib/kafka/client.rb]: W

File: lib/kafka/consumer.rb [hunk 16]
```diff
@@ -186,11 +186,11 @@ def paused?(topic, partition)
     # @param automatically_mark_as_processed [Boolean] whether to automatically
     #   mark a message as successfully processed when the block returns
     #   without an exception. Once marked successful, the offsets of processed
-    #   messages can be committed to Kafka.
-    # @yieldparam message [Kafka::FetchedMessage] a message fetched from Kafka.
-    # @raise [Kafka::ProcessingError] if there was an error processing a message.
+    #   messages can be committed to EbKafka.
+    # @yieldparam message  [EbKafka::FetchedMessage] a message fetched from EbKafka.
```
Comment (echistyakov) [inline lib/kafka/consumer.rb]: W

File: lib/kafka/producer.rb [hunk 17]
```diff
@@ -6,17 +6,17 @@
...
 require "kafka/compressor"
 
-module Kafka
+module EbKafka
 
-  # Allows sending messages to a Kafka cluster.
+  # Allows sending messages to a EbKafka cluster.
   #
-  # Typically you won't instantiate this class yourself, but rather have {Kafka::Client}
+  # Typically you won't instantiate this class yourself, but rather have  {EbKafka::Client}
```
Comment (echistyakov) [inline lib/kafka/producer.rb]: W

**Review by echistyakov — APPROVED** [review]

**Pull request merged**